fdbserver/ptxn/test/Driver.actor.cpp

```diff
@@ -0,0 +1,185 @@
+		actors.emplace_back(fakeProxy(pFakeProxyContext));
+	}
+}
+
+void startFakeTLog(std::vector<Future<Void>>& actors, std::shared_ptr<TestDriverContext> pTestDriverContext) {
+	for (int i = 0; i < pTestDriverContext->numTLogs; ++i) {
+		std::shared_ptr<FakeTLogContext> pFakeTLogContext(
+		    new FakeTLogContext{ pTestDriverContext, pTestDriverContext->tLogInterfaces[i] });
+		actors.emplace_back(getFakeTLogActor(pTestDriverContext->messageTransferModel, pFakeTLogContext));
+	}
+}
+
+void startFakeStorageServer(std::vector<Future<Void>>& actors, std::shared_ptr<TestDriverContext> pTestDriverContext) {
+	for (int i = 0; i < pTestDriverContext->numStorageServers; ++i) {
+		std::shared_ptr<FakeStorageServerContext> pFakeStorageServerContext(
+		    new FakeStorageServerContext{ pTestDriverContext, pTestDriverContext->storageServerInterfaces[i] });
+		actors.emplace_back(
+		    getFakeStorageServerActor(pTestDriverContext->messageTransferModel, pFakeStorageServerContext));
+	}
+}
+
+void printCommitRecord(const std::vector<CommitRecord>& records) {
+	std::cout << "Commits from Proxy: \n\n";
+	Version currentVersion = 0;
+	for (const auto& record : records) {
+		if (record.version != currentVersion) {
+			std::cout << "\n\tVersion: " << record.version << "\n\n";
+			currentVersion = record.version;
+		}
+		std::cout << "\t\tTeam ID: " << record.teamID.toString() << std::endl;
+		for (const auto& mutation : record.mutations) {
+			std::cout << "\t\t\t" << mutation.toString() << std::endl;
+		}
+	}
+}
+
+void verifyMutationsInRecord(const std::vector<CommitRecord>& records,
+                             const Version& version,
+                             const TeamID& teamID,
+                             const std::vector<MutationRef>& mutations) {
+	for (auto& record : records) {
+		if (record.version == version && record.teamID == teamID && record.mutations.size() == mutations.size()) {
+			bool isSame = true;
+			for (size_t i = 0; i < mutations.size(); ++i) {
+				if (!(record.mutations[i].type == mutations[i].type &&
+				      record.mutations[i].param1 == mutations[i].param1 &&
+				      record.mutations[i].param2 == mutations[i].param2)) {
+					isSame = false;
+					break;
+				}
+			}
+			if (isSame)
+				return;
+		}
+	}
+
+	printCommitRecord(records);
+	std::cout << "\n\nLooking for: Version " << version << "\tTeam ID: " << teamID.toString() << "\n";
+	for (const auto& mutation : mutations) {
+		std::cout << "\t\t\t" << mutation.toString() << std::endl;
+	}
+
+	throw internal_error_msg("Mutations does not match previous record");
+}
+
+} // namespace ptxn
+
+TEST_CASE("/fdbserver/ptxn/test/driver") {
+	using namespace ptxn;
+
+	std::vector<Future<Void>> actors;
+
+	std::shared_ptr<TestDriverContext> context = initTestDriverContext();
+
+	startFakeProxy(actors, context);
+	startFakeTLog(actors, context);
+	startFakeStorageServer(actors, context);
+
+	wait(quorum(actors, 1));
```

Comment:
As we discussed, please add an AsyncVar so that proxy only exists when TLogs and StorageServers have verified the data.